File: official/docs/java/current/carbon-offset/one-call-buy.java
package carbon_offset;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;
import com.easypost.model.Shipment;

import java.util.HashMap;

public class OneCallBuy {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        HashMap<String, Object> toAddressMap = new HashMap<String, Object>();
        HashMap<String, Object> fromAddressMap = new HashMap<String, Object>();
        HashMap<String, Object> parcelMap = new HashMap<String, Object>();
        HashMap<String, Object> shipmentMap = new HashMap<String, Object>();

        shipmentMap.put("carrier_accounts", "ca_...");
        shipmentMap.put("service", "NextDayAir");
        shipmentMap.put("to_address", toAddressMap);
        shipmentMap.put("from_address", fromAddressMap);
        shipmentMap.put("parcel", parcelMap);

        Shipment shipment = Shipment.create(shipmentMap, true);

        System.out.println(shipment);
    }
}

File: official/docs/php/current/carbon-offset/one-call-buy.php
<?php

\EasyPost\EasyPost::setApiKey($_ENV['EASYPOST_API_KEY']);

$toAddress = \EasyPost\Address::create(...);
$fromAddress = \EasyPost\Address::create(...);
$parcel = \EasyPost\Parcel::create(...);

// Reusing Easypost Objects

$shipment = \EasyPost\Shipment::create(
    [
        "carrier_accounts" => ["ca_..."],
        "service" => "NextDayAir",
        "to_address" => $toAddress,
        "from_address" => $from_address,
        "parcel" => $parcel
    ],
    null,
    true
);
